mmrotate/datasets/hrsc.py

```diff
@@ -0,0 +1,256 @@
+            else:
+                data_info['ann']['bboxes_ignore'] = np.zeros((0, 5),
+                                                             dtype=np.float32)
+                data_info['ann']['labels_ignore'] = np.array([],
+                                                             dtype=np.int64)
+                data_info['ann']['polygons_ignore'] = np.zeros(
+                    (0, 8), dtype=np.float32)
+                data_info['ann']['headers_ignore'] = np.zeros((0, 2),
+                                                              dtype=np.float32)
+
+            data_infos.append(data_info)
+        return data_infos
+
+    def _filter_imgs(self):
+        """Filter images without ground truths."""
+        valid_inds = []
+        for i, data_info in enumerate(self.data_infos):
+            if data_info['ann']['labels'].size > 0:
+                valid_inds.append(i)
+        return valid_inds
+
+    def evaluate(self,
+                 results,
+                 metric='mAP',
+                 logger=None,
+                 proposal_nums=(100, 300, 1000),
+                 iou_thr=0.5,
+                 scale_ranges=None,
+                 use_07_metric=True,
+                 nproc=4):
+        """Evaluate the dataset.
+
+        Args:
+            results (list): Testing results of the dataset.
+            metric (str | list[str]): Metrics to be evaluated.
+            logger (logging.Logger | None | str): Logger used for printing
+                related information during evaluation. Default: None.
+            proposal_nums (Sequence[int]): Proposal number used for evaluating
+                recalls, such as recall@100, recall@1000.
+                Default: (100, 300, 1000).
+            iou_thr (float | list[float]): IoU threshold. It must be a float
+                when evaluating mAP, and can be a list when evaluating recall.
+                Default: 0.5.
+            scale_ranges (list[tuple] | None): Scale ranges for evaluating mAP.
+                Default: None.
+            use_07_metric (bool): Whether to use the voc07 metric.
+            nproc (int): Processes used for computing TP and FP.
+                Default: 4.
+        """
+        if not isinstance(metric, str):
+            assert len(metric) == 1
+            metric = metric[0]
+        allowed_metrics = ['mAP', 'recall']
+        if metric not in allowed_metrics:
+            raise KeyError(f'metric {metric} is not supported')
+
+        annotations = [self.get_ann_info(i) for i in range(len(self))]
+        eval_results = OrderedDict()
+        iou_thrs = [iou_thr] if isinstance(iou_thr, float) else iou_thr
+        if metric == 'mAP':
+            assert isinstance(iou_thrs, list)
+            mean_aps = []
+            for iou_thr in iou_thrs:
+                print_log(f'\n{"-" * 15}iou_thr: {iou_thr}{"-" * 15}')
+                mean_ap, _ = eval_map(
+                    results,
+                    annotations,
```

Comment:
The name of `eval_map` has been changed to `eval_rbbox_map`.

Reply:
fix